mriqc/tests/test_parser.py

```diff
@@ -0,0 +1,98 @@
+# emacs: -*- mode: python; py-indent-offset: 4; indent-tabs-mode: nil -*-
+# vi: set ft=python sts=4 ts=4 sw=4 et:
+#
+# Copyright 2021 The NiPreps Developers <nipreps@gmail.com>
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# We support and encourage derived works from this project, please read
+# about our expectations at
+#
+#     https://www.nipreps.org/community/licensing/
+#
+
+"""Test parser."""
+
+import pytest
+
+from mriqc.cli.parser import _build_parser
+
+MIN_ARGS = ['bids_dir', 'out', 'participant']
+
+
+@pytest.mark.parametrize(
+    ('args', 'code'),
+    [
+        ([], 2),
+    ],
+)
+def test_parser_errors(args, code):
+    """Check behavior of the parser."""
+    with pytest.raises(SystemExit) as error:
+        _build_parser().parse_args(args)
+
+    assert error.value.code == code
+
+
+@pytest.mark.parametrize(
+    'args',
+    [
+        MIN_ARGS,
+    ],
+)
+def test_parser_valid(tmp_path, args):
+    """Check valid arguments."""
+    datapath = tmp_path / 'data'
+    datapath.mkdir(exist_ok=True)
+    args[0] = str(datapath)
+
+    opts = _build_parser().parse_args(args)
+
+    assert opts.bids_dir == datapath
+
+
+@pytest.mark.parametrize(
+    ('argval', '_verbosity'),
+    [
+        ('v', 'v'),
+        ('vv', 'vv'),
+        ('vvv', 'vvv'),
+    ],
+)
+def test_verbosity_arg(tmp_path, argval, _verbosity):
```

Comment:
Although I like the cleverness of the proposal, I preferred to keep the verbosity and species test as separate methods to make things more easily readable. Also, because then one could wonder why we don't use a single test for all parameters.

Reply:
The test is passing:
https://app.circleci.com/pipelines/github/nipreps/mriqc/1367/workflows/be694492-cf27-4a77-bc33-35ab7b271067/jobs/8273

Resolving.